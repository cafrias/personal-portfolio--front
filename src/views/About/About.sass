@import '../../_sass/_constants.sass'
@import '../../_sass/_breakpoints.sass'

.about
  display: flex
  flex-direction: column
  align-items: center
  width: 100%
  padding: 0.5rem 0.75rem

.about__heading
  width: 100%

.article-heading::before
  display: block
  content: ""
  margin: 0 auto 1.125rem auto
  width: .5rem
  height: 5.125rem
  border-left: 1px solid $mid-grey

+for-tablet-portrait-up
  .about
    padding: 0
