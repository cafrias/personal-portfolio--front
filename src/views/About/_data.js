// @flow

// LOGOS
import frrgLogo from './_assets/frrg.svg'
import fishing from './_assets/fishing.svg'
import omLogo from './_assets/om.svg'
import mongoLogo from '../../_assets/mongo.svg'
import pianoLogo from './_assets/piano.svg'
import pizzaLogo from './_assets/pizza.svg'
import udacityLogo from './_assets/udacity.svg'
import zenLogo from './_assets/zen.svg'

export const education = [
  {
    logo: udacityLogo,
    sub: 'Front-end',
    h: 'Nanodegree',
    text: 'Udacity<br/>2016'
  },
  {
    logo: mongoLogo,
    sub: 'MongoDB',
    h: 'Certified Developer',
    text: 'MongoDB<br/>2016'
  },
  {
    logo: frrgLogo,
    sub: 'Computer Programming',
    h: 'Associate\'s degree',
    text: 'Facultad Regional Río Grande (FRRG) de la UTN<br/>2013-2015'
  },
]

export const interests = [
  {
    logo: zenLogo,
    sub: 'Head-space',
    h: 'Zen Buddhism',
    text:
      'I’ve been interested in Zen Buddhism since my trip to Japan, after visiting Soji Temple in Yokohama … so peaceful!'
  },
  {
    logo: fishing,
    sub: 'Outdoors',
    h: 'Fishing',
    text: 'Here in Rio Grande we take Fly Fishing seriously. When the season is off, I tie my own flies and also go surf fishing on the beach'
  },
  {
    logo: omLogo,
    sub: 'Exercises',
    h: 'Yoga',
    text:
      'I practiced martial arts for several years when I was a teenager, one day I give Yoga a go, and ever since, I practice at least a little every day.'
  },
  {
    logo: pianoLogo,
    sub: 'Arts',
    h: 'Piano',
    text:
      'I’ve playing piano for several years, it used to be more than a hobby in the past. Nowadays it’s there for eventual diminished scale smashing.'
  },
  {
    logo: pizzaLogo,
    sub: 'Food',
    h: 'Pizza',
    text: 'Ah! Pizza!'
  }
]
