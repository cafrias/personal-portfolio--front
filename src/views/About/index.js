// @flow

import React from 'react'

// ASSETS ______________________________________________________________________
import './About.sass'
import heading from './_assets/heading.svg'

// COMPONENTS __________________________________________________________________
import ArticleLine from '../../components/ArticleLine'
import Filter from '../../components/Filter'

// DATA ________________________________________________________________________
import {education, interests} from './_data'

const About = (props: any) => (
  <main className="about">
    <Filter type="top"/>
    <Filter type="bottom"/>
    <header className="section-header about__header">
      <h1 className="section__heading"
        aria-label="About"
      >
        <img src={heading} alt="About" className="about__heading"/>
      </h1>
    </header>
    <article className="bio">
      <h2 className="article-heading">Bio</h2>
      <p>
        Straight from the very south of Argentina, Rio Grande, I grew up to become a <strong>COMPUTER GEEK</strong> (proudly). Since my childhood, I’ve been around computers and other small devices, like watches, as well.
      </p>
      <p>
        One may think that I would have become a developer rightaway, but eventually I started my college days as Musical Arts Student, and became a lover of Romantic piano composers like Franz Listz and Frederic Chopin. After experimenting with Jazz Piano, I played with a Jazz Quartet and on my own, until I later discovered that music, although my passion, wasn't something I wanted to live of.
      </p>
      <p>
        After a couple of years of experimenting, I went back to my roots and started studying <strong>Computer Programming</strong>, and finally found my vocation. Having a background on Arts motivated me to experiment with <strong>Graphic Design</strong> as well, that way I landed into <strong>UX/UI Design</strong>, which is a core aspect of my life alongside <strong>Full-stack Development</strong>.
      </p>
      <p>
        In the present day, I want to focus all my time to <strong>deliver Great Experiences and Usable Products</strong>, that doesn’t only cover the mandatory guidelines, but also give an extra value to users, clients, designers and developers, through innovation.
      </p>
      <p>
        I'm also commited to accessiblity, I deeply believe in a <strong>Web accessible for all</strong>, without any kind of discrimination (conscious or not); with that in mind I carefully craft and test my projects for ensuring accessiblity.
      </p>
    </article>
    <ArticleLine
      heading="Education"
      entries={education}
    />
    <ArticleLine
      heading="Interests"
      entries={interests}
    />
  </main>
)

export default About
