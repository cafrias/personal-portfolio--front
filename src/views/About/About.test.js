import React from 'react'
import {shallow} from 'enzyme'

import About from './index'

describe('About', () => {
  const wrapper = shallow(<About />)

  it('should render without crashing', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
