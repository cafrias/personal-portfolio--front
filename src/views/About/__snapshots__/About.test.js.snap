exports[`About should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <main
    className="page-template">
    <Connect(Component)
        direction="up"
        label="Home"
        ready={false} />
    <Connect(Filter)
        type="top" />
    <Connect(Filter)
        type="bottom" />
    <header
        className="section-header w-100">
        <PopIn
            onRest={[Function]}>
            <h1
                aria-label="About"
                className="section__heading">
                <img
                    alt="About"
                    className="img__heading"
                    src="heading.svg" />
            </h1>
        </PopIn>
    </header>
    <About
        ready={false} />
    <Connect(Component)
        direction="down"
        label="Skills"
        ready={false} />
</main>,
  "nodes": Array [
    <main
      className="page-template">
      <Connect(Component)
            direction="up"
            label="Home"
            ready={false} />
      <Connect(Filter)
            type="top" />
      <Connect(Filter)
            type="bottom" />
      <header
            className="section-header w-100">
            <PopIn
                  onRest={[Function]}>
                  <h1
                        aria-label="About"
                        className="section__heading">
                        <img
                              alt="About"
                              className="img__heading"
                              src="heading.svg" />
                  </h1>
            </PopIn>
      </header>
      <About
            ready={false} />
      <Connect(Component)
            direction="down"
            label="Skills"
            ready={false} />
</main>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PageTemplate />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PageTemplate {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "titleAnimationEnd": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <main
          className="page-template">
          <Connect(Component)
                    direction="up"
                    label="Home"
                    ready={false} />
          <Connect(Filter)
                    type="top" />
          <Connect(Filter)
                    type="bottom" />
          <header
                    className="section-header w-100">
                    <PopIn
                              onRest={[Function]}>
                              <h1
                                        aria-label="About"
                                        className="section__heading">
                                        <img
                                                  alt="About"
                                                  className="img__heading"
                                                  src="heading.svg" />
                              </h1>
                    </PopIn>
          </header>
          <About
                    ready={false} />
          <Connect(Component)
                    direction="down"
                    label="Skills"
                    ready={false} />
</main>,
        "_debugID": 2,
        "_renderedOutput": <main
          className="page-template">
          <Connect(Component)
                    direction="up"
                    label="Home"
                    ready={false} />
          <Connect(Filter)
                    type="top" />
          <Connect(Filter)
                    type="bottom" />
          <header
                    className="section-header w-100">
                    <PopIn
                              onRest={[Function]}>
                              <h1
                                        aria-label="About"
                                        className="section__heading">
                                        <img
                                                  alt="About"
                                                  className="img__heading"
                                                  src="heading.svg" />
                              </h1>
                    </PopIn>
          </header>
          <About
                    ready={false} />
          <Connect(Component)
                    direction="down"
                    label="Skills"
                    ready={false} />
</main>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PageTemplate />,
}
`;
